util/DirectorySourceQueue.cpp

```diff
@@ -360,7 +366,9 @@ bool DirectorySourceQueue::explore() {
               !std::regex_match(newRelativePath, includeRegex)) {
             continue;
           }
-          WdtFileInfo fileInfo(newRelativePath, fileStat.st_size, directReads_);
+          int perm = fileStat.st_mode & (S_IRWXU | S_IRWXG | S_IRWXO);
```

Comment:
We need this logic also in enqueueFiles() method. If the users provide a list of files, we use that method to populate the queue.